### PR TITLE
chore(systemd): Make systemd service file configurable

### PR DESCRIPTION
**Pull request by lwander**

### CODE REVIEW

**Comment by lwander** [comment]
I published Halyard with a `xenial` distribution enabled which didn't work since we don't have any systemd service files. Supplying this gradle update allows us to configure systemd settings for all our services.

File: src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy [hunk 1]
```diff
@@ -79,6 +79,16 @@ class SpinnakerPackagePlugin implements Plugin<Project> {
                 setFileType(new Directive(Directive.RPMFILE_CONFIG | Directive.RPMFILE_NOREPLACE))
             }
         }
+        
+        def systemdService = project.file("lib/systemd/system/${appName}.service")
```
Comment (jtk54) [inline src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy]: These will need added to each microservice, correct?

File: src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy [hunk 2]
```diff
@@ -79,6 +79,16 @@ class SpinnakerPackagePlugin implements Plugin<Project> {
...
+        if (systemdService.exists()) {
+            extension.from(systemdService) {
+                into('/lib/systemd/system')
```
Comment (jtk54) [inline src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy]: Does this affect Upstart on 14.04 if these files are present? How different are the Upstart/systemd configs?

File: src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy [hunk 3]
```diff
@@ -79,6 +79,16 @@ class SpinnakerPackagePlugin implements Plugin<Project> {
...
+                setUser('root')
+                setPermissionGroup('root')
+                setFileType(new Directive(Directive.RPMFILE_CONFIG | Directive.RPMFILE_NOREPLACE))
```
Comment (jtk54) [inline src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy]: I know this is copypasta from the upstart config, but why isn't there a `DEBFILE_CONFIG` etc? @cfieber

**Comment by jtk54** [comment]
Are there any other gradle changes necessary to produce a `xenial` distribution? Do we need to add defaults to https://github.com/spinnaker/spinnaker-gradle-project/blob/master/src/main/groovy/com/netflix/spinnaker/gradle/project/SpinnakerProjectConventionsPlugin.groovy#L68?

File: src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy (hunk 1 above)
Comment (lwander) [inline src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy]: Yup

File: src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy (hunk 2 above)
Comment (lwander) [inline src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy]: 14.04 doesn't use systemd

they are different init systems, with different config languages

File: src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy (hunk 3 above)
Comment (cfieber) [inline src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy]: don't know if the equivalent directives exist to flag files in a deb as config or not (and if so whether they are supported in the gradle plugin)

**Pull request merged**